numpy/core/src/multiarray/refcount.c

```diff
@@ -21,6 +25,94 @@ static void
 _fillobject(char *optr, PyObject *obj, PyArray_Descr *dtype);
 
 
+/*
+ * Helper function to clear a strided memory (normally or always contiguous)
+ * from all Python (or other) references.  The function does nothing if the
+ * array dtype does not indicate holding references.
+ *
+ * It is safe to call this function more than once, failing here is usually
+ * critical (during cleanup) and should be set up to minimize the risk or
+ * avoid it fully.
+ */
+NPY_NO_EXPORT int
+PyArray_ClearBuffer(
+        PyArray_Descr *descr, char *data,
+        npy_intp stride, npy_intp size, int aligned)
+{
+    if (!PyDataType_REFCHK(descr)) {
+        return 0;
+    }
+
+    NPY_traverse_info clear_info;
+    NPY_ARRAYMETHOD_FLAGS flags;
+    if (PyArray_GetClearFunction(
+            aligned, stride, descr, &clear_info, &flags) < 0) {
+        return -1;
+    }
+
+    int res = clear_info.func(
+            NULL, clear_info.descr, data, size, stride, clear_info.auxdata);
+    NPY_traverse_info_xfree(&clear_info);
```

Comment:
What are the flags for?

Reply:
Future proofing and symmetry.  The only flags we have are requiring GIL and floating point errors.  Neither are relevant for clearing I think (I guess releasing GIL could be in principle, so maybe I could add a check for that).

I hope that the pattern could be used e.g. for sorting where flags would be more meaningful.